02-tamagotchi-interaction/io/src/lib.rs

```diff
@@ -2,34 +2,97 @@
 
 use codec::{Decode, Encode};
 use gmeta::Metadata;
-use gstd::prelude::*;
+use gstd::{ActorId, prelude::*};
 use scale_info::TypeInfo;
+use gmeta::{In, InOut};
 
-#[derive(Default, Encode, Decode, TypeInfo)]
+pub const HUNGER_PER_BLOCK: u64 = 1;
+pub const BOREDOM_PER_BLOCK: u64 = 2;
+pub const ENERGY_PER_BLOCK: u64 = 2;
+pub const FILL_PER_FEED: u64 = 1000;
+pub const FILL_PER_ENTERTAINMENT: u64 = 1000;
+pub const FILL_PER_SLEEP: u64 = 1000;
+
+#[derive(Clone, Default, Encode, Decode, TypeInfo, Debug)]
 pub struct Tamagotchi {
-    // TODO: 0️⃣ Copy fields from previous lesson and push changes to the master branch
-    // TODO: 1️⃣ Add new fields
+    pub name: String,
+    pub date_of_birth: u64,
+    pub owner: ActorId,
+    pub fed: Mood,
+    pub fed_block: u64,
+    pub entertained: Mood,
+    pub entertained_block: u64,
+    pub slept: Mood,
+    pub slept_block: u64,
+}
+
+#[derive(Clone, Encode, Decode, TypeInfo, Debug)]
+pub struct Mood(u64);
+
+impl Mood {
+    pub fn try_add(&mut self, n: u64) -> Result<(), ()> {
+        let tmp = self.0 + n;
+        if tmp < 10_000 {
```

Comment:
Using a named constant for `10_000` is better as it is used multiple times in the code.